inject-java/src/main/java/io/micronaut/annotation/processing/visitor/AbstractJavaElement.java

```diff
@@ -76,7 +77,20 @@
                 .newAnnotationBuilder()
                 .annotate(annotationMetadata, av);
 
-        String declaringTypeName = this instanceof MemberElement ? ((MemberElement) this).getOwningType().getName() : getName();
+        String declaringTypeName;
+        if (this instanceof MemberElement) {
+            declaringTypeName = ((MemberElement) this).getOwningType().getName();
+        } else if (this instanceof ParameterElement) {
+            TypeElement typeElement = visitorContext.getModelUtils().classElementFor((Element) this.getNativeType());
```

Comment:
The only part I am not sure about is if this cast is safe, or if there is another way to obtain the declaring type...

Reply:
Yes it is safe